diff --git a/cli/dist/preconstruct-cli-cli.cjs.js b/cli/dist/preconstruct-cli-cli.cjs.js
index 8fbd4efc8ee4d98bc72d16344ff6f6ab4e5e2736..cf451d99128fd06053426d1a265f4e305380dd7b 100644
--- a/cli/dist/preconstruct-cli-cli.cjs.js
+++ b/cli/dist/preconstruct-cli-cli.cjs.js
@@ -1509,7 +1509,7 @@ let memoizedGetProgram = weakMemoize(typescript => memoize(async configFileName
   return nonMemoizedGetProgram(typescript, configFileName);
 }));
 async function getProgram(dirname, pkgName, ts) {
-  let configFileName = ts.findConfigFile(dirname, ts.sys.fileExists);
+  let configFileName = ts.findConfigFile(dirname, ts.sys.fileExists, 'tsconfig.lib.json');
   if (!configFileName) {
     throw new FatalError("an entrypoint source file ends with the .ts or tsx extension but no TypeScript config exists, please create one.", pkgName);
   }
diff --git a/cli/dist/preconstruct-cli-cli.esm.js b/cli/dist/preconstruct-cli-cli.esm.js
index 0045392b46c8928753af564f67681886e0f15bed..164a9bfbae85f1a8f41fffc87c6a3c4014cb9d33 100644
--- a/cli/dist/preconstruct-cli-cli.esm.js
+++ b/cli/dist/preconstruct-cli-cli.esm.js
@@ -1460,7 +1460,7 @@ let memoizedGetProgram = weakMemoize(typescript => memoize(async configFileName
   return nonMemoizedGetProgram(typescript, configFileName);
 }));
 async function getProgram(dirname, pkgName, ts) {
-  let configFileName = ts.findConfigFile(dirname, ts.sys.fileExists);
+  let configFileName = ts.findConfigFile(dirname, ts.sys.fileExists, 'tsconfig.lib.json');
   if (!configFileName) {
     throw new FatalError("an entrypoint source file ends with the .ts or tsx extension but no TypeScript config exists, please create one.", pkgName);
   }
