diff --git a/index.js b/index.js
index d518f1df0974b361911ab5ed90857295ab834f0a..919937ce987cf89fb5718837970406d1cd5bfc4b 100644
--- a/index.js
+++ b/index.js
@@ -59,7 +59,7 @@ function parse (string, allowUnsafe) {
     }
   }
 
-  var loader = allowUnsafe ? parser.load : parser.safeLoad
+  var loader = parser.load
   var yaml = match[match.length - 1].replace(/^\s+|\s+$/g, '')
   var attributes = loader(yaml) || {}
   var body = string.replace(match[0], '')
